converter/src/main/java/de/ubleipzig/iiifproducer/converter/StructureBuilderVersion3.java

```diff
@@ -0,0 +1,304 @@
+                            List<String> rangeLabels = structureLabels.get();
+                            if (!rangeLabels.isEmpty()) {
+                                final Map<String, List<String>> rangeLabelMap = new HashMap<>();
+                                rangeLabelMap.put(DEUTSCH, rangeLabels);
+                                nr.setLabel(rangeLabelMap);
+                            }
+                        }
+                        final Optional<List<MetadataVersion3>> structureMetadata = ofNullable(
+                                buildStructureMetadataForId(sId));
+                        if (structureMetadata.isPresent()) {
+                            List<MetadataVersion3> m = structureMetadata.get();
+                            if (!m.isEmpty()) {
+                                nr.setMetadata(m);
+                            }
+                        }
+                        newRanges.add(nr);
+                }
+            }
+            final Stream<Item> combinedItems = Stream.concat(newRanges.stream(), newCanvases.stream());
+            final List<Item> newItems = combinedItems.collect(Collectors.toList());
+            newStructure.setItems(newItems);
+            final String structId = struct.getId();
+            newStructure.setId(structId);
+            final String sId;
+            try {
+                sId = new URL(structId).getPath().split(separator)[3];
+               final Optional<List<MetadataVersion3>> structureMetadata = ofNullable(
+                        buildStructureMetadataForId(sId));
+                if (structureMetadata.isPresent()) {
+                    List<MetadataVersion3> m = structureMetadata.get();
+                    if (!m.isEmpty()) {
+                        structureMetadata.ifPresent(newStructure::setMetadata);
+                    }
+                }
+
+                final Optional<List<String>> structureLabels = ofNullable(
+                        buildStructureLabelsForId(sId));
+                if (structureLabels.isPresent()) {
+                    final Map<String, List<String>> labelMap = new HashMap<>();
+                    List<String> m = structureLabels.get();
+                    if (!m.isEmpty()) {
+                        labelMap.put(DEUTSCH, m);
+                        newStructure.setLabel(labelMap);
+                    }
+                }
+            } catch (MalformedURLException e) {
+                e.printStackTrace();
+            }
+            newStructures.add(newStructure);
+        }
+        //finally merge all substructures into top structure
+        final Optional<Item> topStructure = newStructures.stream().filter(
+                s -> s.getId().contains("LOG_0000")).findAny();
+        if (topStructure.isPresent()) {
+            final Map<String, List<String>> topStructurelabelMap = buildLabelMap("Contents", ENGLISH);
+            topStructure.get().setLabel(topStructurelabelMap);
```

Comment:
Converting the v2 top level structure (which is usually not displayed by clients) into an own structure creates an additional navigation entry in both Universal Viewer and Mirador 3. Both viewers could display manifests without such a top level Range correctly, but creating a top level structure is in what's suggested in the [v3 cookbook recipe](https://iiif.io/api/cookbook/recipe/0024-book-4-toc/), so I'm not sure how to proceed.

For now, it might be sufficient to add an additional label/value pair with for German and "Inhalt".

![Mirador 3](https://user-images.githubusercontent.com/20013890/156564400-0f54d7e8-e6d3-46d7-aa65-ce1a8ee498d9.png)
![Universal Viewer](https://user-images.githubusercontent.com/20013890/156564405-59d0e156-75f1-4f4d-9fcf-84ba1a9a1f1a.png)

Reply:
implemented "de" label ("Inhalt") for top structure as suggested
![Screen Shot 2022-03-09 at 9 05 34 AM (3)](https://user-images.githubusercontent.com/5990301/157457452-c094a102-c385-4c72-8d87-ec1d8b89f332.png)